HSLdevcom/jore-map-ui
# 603 create link tool

Closes #603 

Backend PR: https://gitlab.hsl.fi/jore/jore-map-backend/merge_requests/35

* DialogStore's setFadeMessage now returns a promise so that redirect to /link/... can be made after 'Tallennettu' text disappears

* Added comment to navigator "prevent new pushing url if the current rl is already the right one"
    * this happened if url was already new/link and user created a new link

* sometimes when creating new link it didn't work at the first attempt. Couldn't reproduce this error anymore though.

* Didn't want to refactor sending whole node when saving ILink
   * added 2 TODOs: "TODO: refactor as startNodeId? TODO: refactor as endNodeId?"

New tasks:

* Save button is now disabled if link from selected startNode --> endNode exist with the current transitType. There should be some visualization for the user of what is wrong, card -> #747 

* Create link / node: fill in address information automatically. Card -> #736 

Test instructions:

* Test that you can't create links with existing transitType + start & end node
   * This is not however prevented in backend -> Save should throw error if foreign constraint is broken
   * Added a test card for link creation: #749

* Test that back button works after link creation

* Test updating existing link

etc...
